src/pk/ecc/ecc_sign_hash_rfc7518.c

```diff
@@ -0,0 +1,83 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt_private.h"
+
+#ifdef LTC_MECC
+
+/**
+  Sign a message digest (RFC7518 format + recovery_id)
+  @param in        The message digest to sign
+  @param inlen     The length of the digest
+  @param out       [out] The destination for the signature
+  @param outlen    [in/out] The max size and resulting size of the signature
+  @param prng      An active PRNG state
+  @param wprng     The index of the PRNG you wish to use
+  @param recid     [out] Recovery ID
+  @param key       A private ECC key
+  @return CRYPT_OK if successful
+*/
+int ecc_sign_hash_rfc7518_ex(const unsigned char *in,  unsigned long inlen,
+                             unsigned char *out, unsigned long *outlen,
+                             prng_state *prng, int wprng,
+                             int *recid, const ecc_key *key)
+{
+   int err;
+   void *r, *s;
+   unsigned long pbytes, i;
+
+   LTC_ARGCHK(out    != NULL);
+   LTC_ARGCHK(outlen != NULL);
+   LTC_ARGCHK(key    != NULL);
+
+   if ((err = mp_init_multi(&r, &s, NULL)) != CRYPT_OK) return err;
+   if ((err = ecc_sign_hash_internal(in, inlen, r, s, prng, wprng, recid, key)) != CRYPT_OK) goto error;
+
+   /* RFC7518 format - raw (r,s) */
+   pbytes = mp_unsigned_bin_size(key->dp.order);
+   if (*outlen < 2 * pbytes) {
+      err = CRYPT_BUFFER_OVERFLOW;
+      *outlen = 2 * pbytes;
+      goto error;
+   }
+   zeromem(out, 2 * pbytes);
+   *outlen = 2 * pbytes;
+   i = mp_unsigned_bin_size(r);
+   if ((err = mp_to_unsigned_bin(r, out + pbytes - i)) != CRYPT_OK) goto error;
+   i = mp_unsigned_bin_size(s);
+   err = mp_to_unsigned_bin(s, out + 2 * pbytes - i);
+
+error:
+   mp_clear_multi(r, s, NULL);
+   return err;
+}
+
+/**
+  Sign a message digest (RFC7518 format)
+  @param in        The message digest to sign
+  @param inlen     The length of the digest
+  @param out       [out] The destination for the signature
+  @param outlen    [in/out] The max size and resulting size of the signature
+  @param prng      An active PRNG state
+  @param wprng     The index of the PRNG you wish to use
+  @param key       A private ECC key
+  @return CRYPT_OK if successful
+*/
+int ecc_sign_hash_rfc7518(const unsigned char *in,  unsigned long inlen,
+                          unsigned char *out, unsigned long *outlen,
+                          prng_state *prng, int wprng, const ecc_key *key)
+{
+   return ecc_sign_hash_rfc7518_ex(in, inlen, out, outlen, prng, wprng, NULL, key);
```

Comment:
why not make `ecc_sign_hash_rfc7518_ex()` the new `ecc_sign_hash_rfc7518`?
we could simply tell the user to pass NULL if he doesn't want the recid... I'd prefer that as the difference is now really not that huge.

Reply:
The `ecc_sign_hash_rfc7518` was already released or not?

Reply:
let me check

Reply:
true, damn, okay let's keep it like that!

Reply:
OTOH will the next release break API anyways... so what!? :)

Reply:
I do not know.

Just a note that there is also #477 which may also need some API changes.

Reply:
ok, let's keep it for now as is